WooCommerce/Classes/ViewRelated/Hub Menu/HubMenu.swift

```diff
@@ -4,13 +4,125 @@ import SwiftUI
 /// and will be the entry point of the `Menu` Tab.
 ///
 struct HubMenu: View {
+    @ObservedObject private var viewModel: HubMenuViewModel
+    @State private var showViewStore = false
+    @State private var showReviews = false
+
+    init(siteID: Int64) {
+        viewModel = HubMenuViewModel(siteID: siteID)
+    }
+
     var body: some View {
-        Text("Work in progress")
+        VStack {
+            TopBar(storeTitle: viewModel.storeTitle,
+                   storeURL: viewModel.storeURL.absoluteString)
+
+            ScrollView {
+                let gridItemLayout = [GridItem(.adaptive(minimum: Constants.itemSize), spacing: Constants.itemSpacing)]
+
+                LazyVGrid(columns: gridItemLayout) {
+                    ForEach(viewModel.menuElements, id: \.self) { menu in
+                        HubMenuElement(image: menu.icon, text: menu.title)
+                            .frame(width: Constants.itemSize, height: Constants.itemSize)
+                            .onTapGesture {
+                                switch menu {
+                                case .viewStore:
+                                    showViewStore = true
+                                case .reviews:
+                                    showReviews = true
+                                default:
+                                    // TODO-5509: handle the remaining cases
+                                    break
+                                }
+                            }
+                    }
+                    .background(Color.white)
+                    .cornerRadius(Constants.cornerRadius)
+                    .padding([.bottom], Constants.padding)
+                }
+                .padding(Constants.padding)
+                .background(Color(.listBackground))
+            }
+            .safariSheet(isPresented: $showViewStore, url: viewModel.storeURL)
+            NavigationLink(destination:
+                            ReviewsView(siteID: viewModel.siteID),
+                           isActive: $showReviews) {
+                EmptyView()
+            }.hidden()
+        }
+        .navigationBarHidden(true)
+        .background(Color(.listBackground).edgesIgnoringSafeArea(.all))
+
+    }
+
+    private struct TopBar: View {
```

Comment:
I'd like to not separate it since it's strictly a component of the main view (it will be not reusable). I just separated it in a different view under the same class because it's more readable.